### PR TITLE
Mobile global view of data globe

### PR DESCRIPTION
**Pull request by aabdaab**
This PR is a starting point for the mobile designs of the platform. It only implements global view styles for data globe.
`responsive.js` util file has been added to facilitate rendering `MobileOnly` components or recognizing if we are on mobile js-way (`isMobile` flag imported is a bunch of files). As the flow on mobile is not exactly the same as on desktop (it's a mix of responsive and reactive approach), some global flags (url) were added to control the experience (i.e. default screen on mobile is without entry boxes open). Note that both of the experiences (desktop and mobile) should be tested separately as not everything is persisted throughout them (for example we do not store the active option of the menu footer in any global state, i.e. going from search widget on desktop to mobile will result in search widget changing styles to mobile BUT the menu footer will not be affected, search option will not be active there.)

New mobile components:
- MenuFooter
- MenuSettings

image

image

image

image

image

### CODE REVIEW

**Comment by kfilip94** [comment]
:eyes: checking

**Comment by kfilip94** [comment]
@aabdaab  below you can find the issues I found so far, I'll take a look at the code now.

- In my opinion the `add layer` and all menu buttons should work like a switch, they should toggle the sub-menus, not only open them. Also problematic is when you want to go back to the only-globe view, because when you will open the menu once, you can not minimize it again. Having the feature like the fullscreen mode here would make sense.
- Menu labels are not centered very well, add top margin or transform property to the css:
image
- Background color of the selected menu option is wrong, it should be `rgba(255,255,255,0.05);`, not `rgba(9,20,26,0.45)`,
- Also color of the icons is wrong, they are white only when they are selected, otherwise they should have opacity `opacity: 0.5;`
- Check the styles of the mobile version of entry boxes, they should have bigger padding on the sides, smaller title font (24px instead of 32px) and only one word per line. I don't know if the arrows and different title font should be there, as they are different in the desktop version.
This gradient line between categories should be 2px height, not 1px.
Journey-game-underground-blue-enviorment
- The arrows next to the Back button are missing, also the left and the right padding of the sidebar in mobile version should be bigger (32px instead of 20px).
image

**Comment by aabdaab** [comment]
I'll try to refactor the flag hell ❌

File: src/components/mobile-only/menu-footer/menu-footer.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+const MenuFooterContainer = props => {
+  const { view, isEntryBoxesOpen, isLegendOpen, isSidebarOpen, isSettingsOpen, isLandscapeMode } = props;
+  const { handleOpenSearch, handleCloseSearch, searchWidget } = useSearchWidgetLogic(
+    view,
+    openPlacesSearchAnalyticsEvent,
+    searchLocationAnalyticsEvent
+  );
+
+  const handleEntryBoxesOpen = () => {if (!isEntryBoxesOpen) props.changeUI({ isEntryBoxesOpen: true }); }
```
[the remark this answers is not in the document]
Comment (aabdaab) [reply src/components/mobile-only/menu-footer/menu-footer.js]: done

**Comment by aabdaab** [comment]
Ready to review again ☝️ 
The missing arrow button in the sidebar is handled in a different PR.

**Pull request merged**